### PR TITLE
DO500 - CRW

### PR DESCRIPTION
**Pull request by springdo**

### CODE REVIEW

**Review by tylerauerbeck — COMMENTED** [review]
@springdo To avoid the workaround, is there a possibility of maybe just pinning CRW to a "healthy version"? I haven't looked too much at the problem, but wherever we end up we probably want to avoid having the operator upgrade things mid-run. So would probably be good to just pick a point to pin to once we land on something that works.

File: codereadyworkspaces/stack/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     echo "⚓️⚓️⚓️⚓️⚓️"
+
+#🦭🦭🦭kubeseal 🦭
+RUN curl -sL https://github.com/bitnami-labs/sealed-secrets/releases/download/${KUBESEAL_VERSION}/kubeseal-linux-amd64 -o /usr/local/bin/kubeseal && \
+    chmod -R 755 /usr/local/bin/kubeseal && \
+    echo "🦭🦭🦭🦭🦭"
+
+# tekton
+RUN curl -sLO https://github.com/tektoncd/cli/releases/download/v${TEKTON_VERSION}/tkn_${TEKTON_VERSION}_Linux_x86_64.tar.gz && \
+    tar xvzf tkn_${TEKTON_VERSION}_Linux_x86_64.tar.gz -C /usr/local/bin/ tkn  && \
```
Comment (eformat) [inline codereadyworkspaces/stack/Dockerfile:33]: this failed for me ? probs the space 🦤 before tkn

may be useful to use the OpenShift matching version from the mirror:
```
ENV TEKTON_VERSION=0.19.1

# tekton
RUN curl -sL https://mirror.openshift.com/pub/openshift-v4/clients/pipeline/latest/tkn-linux-amd64-${TEKTON_VERSION}.tar.gz | tar --strip-components=1 -C /usr/local/bin -xzf - tkn && \
    echo "🐈🐈🐈🐈🐈"
```

File: codereadyworkspaces/stack/centos8.repo [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[base]
+name=CentOS $releasever – Base
+baseurl=http://mirror.centos.org/centos/7/os/$basearch/
```
Comment (eformat) [inline codereadyworkspaces/stack/centos8.repo:3]: can we change this to ubi8.repo ?

```
# base image is UBI
/root/buildinfo/Dockerfile-ubi8-minimal-8.4-205
```

So we could make this:
```
[rhel-8-baseos]
id = rhel-8-baseos-cop
name = Red Hat Universal Base Image 8 Baseos (RPMs)
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi8/8/$basearch/baseos/os
enabled = 1
gpgcheck = 1

[rhel-8-appstream]
id = rhel-8-appstream-cop
name = Red Hat Universal Base Image 8 Appstream (RPMs)
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi8/8/$basearch/appstream/os
enabled = 1
gpgcheck = 1

[rhel-8-codeready-builder-rpms]
id = rhel-8-codeready-builder-rpms-cop
name = Red Hat Universal Base Image 8 Codeready Builder (RPMs)
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi8/8/$basearch/codeready-builder/os
enabled = 1
gpgcheck = 1
```

File: codereadyworkspaces/stack/centos8.repo [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[base]
```
Comment (eformat) [inline codereadyworkspaces/stack/centos8.repo:1]: Lets make this real centos8 stream, rename file to ubi8.repo ?

```
[rhel-8-baseos]
id = rhel-8-baseos-cop
name = Red Hat Universal Base Image 8 Baseos (RPMs)
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi8/8/$basearch/baseos/os
enabled = 1
gpgcheck = 1

[rhel-8-appstream]
id = rhel-8-appstream-cop
name = Red Hat Universal Base Image 8 Appstream (RPMs)
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi8/8/$basearch/appstream/os
enabled = 1
gpgcheck = 1

[rhel-8-codeready-builder-rpms]
id = rhel-8-codeready-builder-rpms-cop
name = Red Hat Universal Base Image 8 Codeready Builder (RPMs)
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi8/8/$basearch/codeready-builder/os
enabled = 1
gpgcheck = 1

# CentOS-Stream-BaseOS.repo
[baseos]
name=CentOS Stream $releasever - BaseOS
baseurl=http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os
gpgcheck=1
enabled=1
```

File: codereadyworkspaces/stack/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    curl -sLo /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_linux_amd64 && \
+    chmod +x /usr/local/bin/yq && \
+    echo "🦨🦨🦨🦨🦨"
+
+# Conftest? This will make Gareth Happy :P
+RUN curl --fail -skL https://github.com/open-policy-agent/conftest/releases/download/v${CONFTEST_VERSION}/conftest_${CONFTEST_VERSION}_Linux_x86_64.tar.gz | tar zxf - -C /usr/local/bin conftest && \
+    echo "🌲🌲🌲🌲🌲"
+
+# zsh
+ADD centos8.repo /etc/yum.repos.d/centos8.repo
```
Comment (eformat) [inline codereadyworkspaces/stack/Dockerfile:48]: see below comment for ubi8.repo and add

```
# UBI8, Centos-Stream8
ADD ubi8.repo /etc/yum.repos.d/ubi8.repo
RUN dnf install -y centos-gpg-keys --nogpgcheck && \
  rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
```

File: codereadyworkspaces/stack/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+# argo
+RUN curl -sL https://github.com/argoproj/argo-cd/releases/download/v${ARGOCD_VERSION}/argocd-linux-amd64 -o /usr/local/bin/argocd && \
+    chmod -R 775 /usr/local/bin/argocd && \
+    echo "🐙🐙🐙🐙🐙"
+
+#helm
+RUN  curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 && \
+     chmod 700 get_helm.sh && \
+     ./get_helm.sh --version ${HELM_VERSION} && \
```
Comment (eformat) [inline codereadyworkspaces/stack/Dockerfile:23]: something funky here ... when i run (with ubi8 changes below) ... helm fails
```
virt:~/git/enablement-framework/codereadyworkspaces ⎇ main#1a20d50$ podman run --rm -it localhost/stack-do500:latest bash
warning: unable to access '/home/jboss/.config/git/config': Permission denied
[1001@4cb49ce667e3 projects]$ zsh
➜  /projects ls
get_helm.sh
➜  /projects 
➜  /projects helm version
WARNING: stat /home/jboss/.config/helm/registry.json: permission denied
/home/jboss/.config/helm/registry.json
github.com/deislabs/oras/pkg/auth/docker.NewClient
	github.com/deislabs/oras@v0.11.1/pkg/auth/docker/client.go:42
helm.sh/helm/v3/internal/experimental/registry.NewClient
	helm.sh/helm/v3/internal/experimental/registry/client.go:70
main.newRootCmd
	helm.sh/helm/v3/cmd/helm/root.go:156
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:70
runtime.main
	runtime/proc.go:225
runtime.goexit
	runtime/asm_amd64.s:1371
```
and may need to rm the get_helm.sh script huh ?

File: codereadyworkspaces/stack/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,67 @@
+FROM registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.10-4
+
+ENV ARGOCD_VERSION=2.0.5
+ENV YQ_VERSION=v4.11.1
+ENV HELM_VERSION=v3.6.0
```
Comment (eformat) [inline codereadyworkspaces/stack/Dockerfile:5]: Use latest helm
```
ENV HELM_VERSION=v3.6.3
```

also .. $HOME does not work with this helm
```
[1001@23ede950920b tmp]$ helm version
WARNING: stat /home/jboss/.config/helm/registry.json: permission denied
```

But
```
[1001@23ede950920b tmp]$ HOME=/tmp
[1001@23ede950920b ~]$ helm version
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"clean", GoVersion:"go1.16.5"}
```
is OK

**Comment by eformat** [comment]
LGTM

**Pull request merged**